x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/components/tables/meta_engines_table.test.tsx

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import '../../../../../__mocks__/enterprise_search_url.mock';
+import { mockTelemetryActions, mountWithIntl, setMockValues } from '../../../../../__mocks__';
+
+import React from 'react';
+
+import { ReactWrapper, shallow } from 'enzyme';
+
+import { EuiBasicTable, EuiPagination, EuiButtonEmpty, EuiIcon, EuiTableRow } from '@elastic/eui';
+
+import { KibanaLogic } from '../../../../../shared/kibana';
+import { EuiLinkTo } from '../../../../../shared/react_router_helpers';
+import { TelemetryLogic } from '../../../../../shared/telemetry';
+import { EngineDetails } from '../../../engine/types';
+import { EnginesLogic } from '../../engines_logic';
+
+import { MetaEnginesTable } from './meta_engines_table';
+import { MetaEnginesTableExpandedRow } from './meta_engines_table_expanded_row';
+import { MetaEnginesTableNameColumnContent } from './meta_engines_table_name_column_content';
+
+describe('MetaEnginesTable', () => {
+  const onChange = jest.fn();
+  const onDeleteEngine = jest.fn();
+
+  const data = [
+    {
+      name: 'test-engine',
+      created_at: 'Fri, 1 Jan 1970 12:00:00 +0000',
+      language: 'English',
+      isMeta: true,
+      document_count: 99999,
+      field_count: 10,
+      includedEngines: [
+        {
+          name: 'source-engine-1',
+        },
+        { name: 'source-engine-2' },
+      ],
+    } as EngineDetails,
+  ];
+  const pagination = {
+    pageIndex: 0,
+    pageSize: 10,
+    totalItemCount: 50,
+    hidePerPageOptions: true,
+  };
+  const props = {
+    items: data,
+    loading: false,
+    pagination,
+    onChange,
+    onDeleteEngine,
+  };
+
+  const DEFAULT_VALUES = {
+    myRole: {
+      canManageMetaEngines: false,
+    },
+    expandedSourceEngines: {},
+    hideRow: jest.fn(),
+    fetchOrDisplayRow: jest.fn(),
+  };
+
+  const resetMocks = () => {
+    jest.clearAllMocks();
+    setMockValues(DEFAULT_VALUES);
+  };
+
+  describe('basic table', () => {
+    let wrapper: ReactWrapper<any>;
+    let table: ReactWrapper<any>;
+
+    beforeAll(() => {
+      resetMocks();
+      wrapper = mountWithIntl(<MetaEnginesTable {...props} />);
+      table = wrapper.find(EuiBasicTable);
+    });
```

Comment:
[Optional] If it's a headache to use beforeAll/manually call `resetMocks()`, I'd just switch to `beforeEach()` instead and let each test handle its own mount. Reducing the # of mounts is a microperf thing I do if it's not a huge pain to do so (e.g. not a lot of props/values to change), but I'm definitely cognizant that isn't the case for this table

Reply:
It makes sense to me to not remount if i'm not expecting differences between tests due to props or state changing, but keeping the nice labelling the test descriptions give us.  Just my two cents.